cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/ActiveOperationRemovalHandler.java

```diff
@@ -91,6 +94,16 @@ public Void call() throws Exception {
         });
       } catch (Exception e) {
         LOG.error("Failed to impersonate while closing handle {}", handle);
+      } finally {
+        try {
+          // we don't want to close FileSystems for the login user
+          if (!UserGroupInformation.getLoginUser().equals(opInfo.getUGI())) {
+            FileSystem.closeAllForUGI(opInfo.getUGI());
```

Comment:
Would be good to add a comment on why this is needed. Also can you add a note to https://issues.cask.co/browse/CDAP-12011 to move this cleanup to happen when the UGI is invalidated from the cache.